### PR TITLE
Add CsvFile.TryFindColumn for #1081

### PR DESCRIPTION
**Pull request by DanielOliver**
I pretty much used the code given in the issue #1081 and added two tests (hopefully in the correct spot).

### CODE REVIEW

**Comment by ovatsus** [comment]
Actually, I just realised there's an existing GetColumnIndex but it's internal. Maybe we would make it public and add this one as TryGetColumnIndex? And I think we should add them only on the non TP CsvFile version

File: tests/FSharp.Data.Tests/CsvProvider.fs [hunk 1]
```diff
@@ -563,3 +563,15 @@ let ``Having null in a cell should not fail saving to string (issue#978)`` () =
     data
     |> Stringify 
     |> ignore
+    
+[<Test>]
+let ``CsvFile.TryFindColumn returns only the first matching column``() =
```
Comment (ovatsus) [inline tests/FSharp.Data.Tests/CsvProvider.fs]: Don't forget to update the test name

Comment (DanielOliver) [reply tests/FSharp.Data.Tests/CsvProvider.fs]: Yep. serves me right for rushing this between things today.

**Comment by ovatsus** [comment]
Thanks

**Pull request merged**